src/main/ApplicationUtils.cpp

```diff
@@ -372,27 +372,16 @@ catchup(Application::pointer app, CatchupConfiguration cc,
         }
         case LedgerManager::LM_SYNCED_STATE:
         {
+            done = true;
+            synced = true;
             break;
         }
         case LedgerManager::LM_CATCHING_UP_STATE:
         {
-            switch (app->getLedgerManager().getCatchupState())
-            {
-            case LedgerManager::CatchupState::WAITING_FOR_CLOSING_LEDGER:
-            {
-                done = true;
-                synced = true;
-                break;
-            }
-            case LedgerManager::CatchupState::NONE:
+            if (app->getLedgerManager().getCatchupState() ==
```

Comment:
`CatchupState::NONE should never occur` in `LM_CATCHING_UP_STATE`. I'll make that an abort() as it would be a logic error.

The splis between catching up states and non-catching up states has two reasons:

1. Before I had to add every LM_CATCHUP_* to every switch statement, and that had no sense at all.

2. The current relationship between `LedgerManager` and `CatchupManager` is broken, and should be fixed (that will require adding a third class that is yet-to-be-named at some point). What I mean is that `LedgerManager` should only operate on catched-up database and do not care about catching up code at all. CatchupState enum should be used only by `CatchupManager`, which should be asking `LedgerManager` to apply ledger for it. A third class should decide where it should send ledgers received from network (to `LedgerManager` if in sync, to `CatchupManager` otherwise). So those two separate enums are there to force us to fix this relationship at some time.